test/cache.test.js

```diff
@@ -0,0 +1,306 @@
+
+import 'mocha';
+import { expect } from 'chai';
+
+import WS from '../src/echo/ws'
+import WSAPI from '../src/echo/ws-api'
+import Cache from '../src/echo/cache'
+import API from '../src/echo/api'
+import { constants } from '../src/index';
+import { USE_CACHE_BY_DEFAULT, DEFAULT_CACHE_EXPIRATION_TIME, DEFAULT_MIN_CACHE_CLEANING_TIME, CACHE_MAPS } from '../src/constants';
+
+import { url } from './_test-data';
+
+describe('cache', () => {
+    describe('options', () => {
+        it('should use default options if cache option has not been specified', () => {
+            try {
+                const cache = new Cache(undefined);
+
+                const { isUsed, expirationTime, minCleaningTime } = cache;
+
+                expect(isUsed).to.be.equal(USE_CACHE_BY_DEFAULT);
+                expect(expirationTime).to.be.equal(DEFAULT_CACHE_EXPIRATION_TIME);
+                expect(minCleaningTime).to.be.equal(DEFAULT_MIN_CACHE_CLEANING_TIME);
+            } catch (e) {
+                throw e;
+            };
+        });
+
+        it('should use default options if empty object was specified', () => {
+            try {
+                const cache = new Cache({});
+
+                const { isUsed, expirationTime, minCleaningTime } = cache;
+
+                expect(isUsed).to.be.equal(USE_CACHE_BY_DEFAULT);
+                expect(expirationTime).to.be.equal(DEFAULT_CACHE_EXPIRATION_TIME);
+                expect(minCleaningTime).to.be.equal(DEFAULT_MIN_CACHE_CLEANING_TIME);     
+            } catch (e) {
+                throw e;
+            }
+        });
+
+        it('should not use the cache if null was specified in the relevant option', () => {
+            try {
+                const cache = new Cache(null);
+
+                const { isUsed } = cache;
+
+                expect(isUsed).to.be.false;
+            } catch (e) {
+                throw e;
+            }
+        });
+
+        it('should use specified options', () => {
+            try {
+                const randomOptions = {
```

Comment:
Random in this case is not a good solution. U need to write 2 tests: one with undefined, one with specific values

Reply:
> Random in this case is not a good solution. U need to write 2 tests: one with undefined, one with specific values

done